http/typescript/fetch/src/fetchOptions.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * @interface {@link https://github.com/bitinn/node-fetch/#options}
+ * Signature to define the fetch request options for node environment
+ * @property {number} [follow] - node-fetch option: maximum redirect count. 0 to not follow redirect
+ * @property {number} [compress] - node-fetch option: support gzip/deflate content encoding. false to disable
+ * @property {number} [size] - node-fetch option: maximum response body size in bytes. 0 to disable
+ * @property {any} [agent] - node-fetch option: HTTP(S).Agent instance, allows custom proxy, certificate, lookup, family etc.
+ * @property {number} [highWaterMark] - node-fetch option: maximum number of bytes to store in the internal buffer before ceasing to read from the underlying resource.
+ * @property {boolean} [insecureHTTPParser] - node-fetch option: use an insecure HTTP parser that accepts invalid HTTP headers when `true`.
+ */
+ export interface NodeFetchInit {
```

Comment:
could we import some type definition instead of redefining the interface?

Reply:
https://github.com/microsoft/kiota/issues/574 Created an issue for this.

Reply:
typically everything you defined is already in request init to the exception of higWaterMark and InsecureHTTPParse: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node-fetch/index.d.ts#L45

Those two are effectively present in node fetch (the javascript code) so maybe we could PR the type definitions and remove this interface? https://github.com/node-fetch/node-fetch/blob/2603c67f1840fb4a43e0837099a3b2725e8f615e/src/request.js#L105